packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -445,6 +446,21 @@ export const RemoteServerFactory = {
         // This method is used by the dev server, but is not needed here.
     },
 
+    /**
+     * Set x-forward-* headers into locals
+     * @private
+     */
+    _setForwardedHeaders(app, options) {
+        app.use((req, res, next) => {
+            const xForwardedHost = req.headers?.['x-forwarded-host']
+            if (xForwardedHost) {
+                res.locals.xForwardedHost = `${options.protocol}://${xForwardedHost}`
```

Comment:
In the context of the HTTP protocol, `origin` and `host` are two different concepts.

`origin`: protocol + hostname + port
`host`: hostname + port

Do we want to keep `res.locals.xForwardedHost` to have the format of a host, i.e. don't add the protocol here

Reply:
Ah yes, good distinction between host and origin.